src/orion/core/worker/knowledge_base.py

```diff
@@ -0,0 +1,73 @@
+""" Interface for the Knowledge Base, which is not currently in the Orion codebase.
```

Comment:
That will need to be changed before we merge. :)

Reply:
Why though? Shouldn't adding the concrete implementation be done in another PR?